components/dropdown/dropdown.tsx

```diff
@@ -118,26 +117,42 @@ export default class Dropdown extends React.Component<DropDownProps, any> {
     return fixedModeOverlay as React.ReactElement;
   };
 
+  getPlacement(direction: string = 'ltr') {
+    const { placement } = this.props;
+    if (placement !== undefined) {
+      return placement;
+    }
+    return direction === 'rtl' ? ('bottomRight' as Placement) : ('bottomLeft' as Placement);
+  }
+
   renderDropDown = ({
     getPopupContainer: getContextPopupContainer,
     getPrefixCls,
+    direction,
   }: ConfigConsumerProps) => {
     const {
       prefixCls: customizePrefixCls,
       children,
       trigger,
       disabled,
       getPopupContainer,
+      overlayClassName,
     } = this.props;
 
     const prefixCls = getPrefixCls('dropdown', customizePrefixCls);
     const child = React.Children.only(children) as React.ReactElement<any>;
 
     const dropdownTrigger = React.cloneElement(child, {
-      className: classNames(child.props.className, `${prefixCls}-trigger`),
+      className: classNames(child.props.className, `${prefixCls}-trigger`, {
+        [`${prefixCls}-rtl`]: direction === 'rtl',
+      }),
       disabled,
     });
 
+    const overlayClassNameCustomized = classNames(overlayClassName, {
```

Comment:
should pass origin `overlayClassName` here.

Reply:
Excuse me, I didn't get it. could you clarify this for me?

Reply:
OPS. wrong file. It's comments here:
https://github.com/ant-design/ant-design/pull/19380/files#diff-186839a30bf8b9d67a4b10bf7c091d5fR261

Reply:
done